src/characters/bob.ts

```diff
@@ -37,11 +35,11 @@ export class RemoteBob {
 }
 
 export class Bob {
-  private readonly porter: Porter;
+  private readonly porter: PorterClient;
   private readonly keyring: Keyring;
 
-  constructor(config: Configuration, secretKey: SecretKey) {
-    this.porter = new Porter(config.porterUri);
+  constructor(porterUri: string, secretKey: SecretKey) {
+    this.porter = new PorterClient(porterUri);
```

Comment:
I don't think Bob should take a Porter uri either - just like Alice.

Reply:
:heavy_check_mark: 

What do you think about other characters like `Decrypter` and `Encrypter`? They are meant to persisted and transferred so it may make sense to store `porterUri` instead of using it ad-hoc.

Reply:
Seems like not. Much like Bob/Alice they *use* a Porter and therefore should probably take a porterUri as a parameter to functions that perform network actions (eg. `retrieve`). 

wdyt, since you are closer to the code than I am?

Reply:
I think it's going to add a little bit of bloat to method signatures, but it may be a good trade-off between ergonomics  and clarity on what consists a protocol object.